src-tauri/src/main.rs

```diff
@@ -32,6 +39,69 @@ fn get_this_binary() -> PathBuf {
   let bin = api::process::current_binary();
   return bin.unwrap();
 }
+#[cfg(target_os = "windows")]
+#[tauri::command]
+fn winreg_get(sub_key: String, value: String) -> String {
+  let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+  let subkey = hkcu.open_subkey(sub_key).unwrap();
+  let val = subkey.get_value(value);
+  let result: String;
+  match val {
+    Ok(content) => {
+      println!("File content: {}", content);
+      result = content;
+    }
+    Err(error) => {
+      println!("There was an error: {}", error);
+      result = error.to_string();
+    }
+  }
+
+  return result;
+}
+
+#[cfg(target_os = "windows")]
+#[tauri::command]
+fn winreg_set(sub_key: String, set_key: String, value: String) -> String {
+  let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+  // let subkey = hkcu.open_subkey(sub_key).unwrap();
+  let (subkey, disp) = hkcu.create_subkey(&sub_key).unwrap();
+  match disp {
+    REG_CREATED_NEW_KEY => println!("A new key has been created"),
+    REG_OPENED_EXISTING_KEY => println!("An existing key has been opened"),
+  }
+  let val = subkey.set_value(set_key, &value);
+  let mut result: String = "success".to_string();
+  match val {
+    Ok(()) => {}
+    Err(error) => {
+      println!("There was an error: {}", error);
+      result = error.to_string();
+    }
+  }
+  return result;
+}
+
+#[cfg(target_os = "windows")]
+#[tauri::command]
+fn winreg_delete(sub_key: String, set_key: String) -> String {
+  let hkcu = RegKey::predef(HKEY_CURRENT_USER);
+  let (subkey, disp) = hkcu.create_subkey(&sub_key).unwrap();
+  match disp {
+    REG_CREATED_NEW_KEY => println!("A new key has been created"),
+    REG_OPENED_EXISTING_KEY => println!("An existing key has been opened"),
+  }
+  let val = subkey.delete_value(set_key);
+  let mut result: String = "success".to_string();
+  match val {
+    Ok(()) => {}
+    Err(error) => {
+      println!("There was an error: {}", error);
+      result = error.to_string();
+    }
+  }
+  return result;
+}
```

Comment:
I would move these 3 into separate module and put one `#[cfg(target_os = "windows")]` on the whole module. That would be cleaner and easier to follow.